### PR TITLE
Extend documentation

### PR DESCRIPTION
**Pull request by dcermak**
This PR proposes some extensions to the documentation.

Changes proposed in this pull request:
* Summary of my pain points & the solution of these when working with KIWI + OBS added to `build_in_buildservice.rst`
* Extension of `vagrant_setup.rst`
* Addition of an "advertising" section to `index.rst` (I can drop this one, not so sure whether this is a good idea)
* Reordering of some TOCs: I have switched the order of a few articles, so that the (imho) more important one appears **first** and thus lazy people (like me) are more likely to read them as I found them more important.
* minor spelling, grammar fixes, whitespace cleanup

### CODE REVIEW

**Comment by schaefi** [comment]
@tomschr if you have a minute to look at the doc changes from Dan it would be great. Thanks much

**Comment by dcermak** [comment]
@tomschr @schaefi I'll proofread the whole thing once more today or on Monday, so don't bother checking this before I mark it as ready for review.

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 1]
```diff
@@ -69,21 +70,40 @@ the following changes are required:
...
+   authorized keys for the user ``vagrant`` so that vagrant itself can
+   connect to the box via ssh.
+   The key can be obtained `GitHub
+   <https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub>`_
+   and should be inserted into the file
+   :file:`home/vagrant/.ssh/authorized_keys`, which can be added as an
+   overlay file into the image description.
 
-5. Setup and start SSH daemon
+   Please keep in mind to set the file system permissions of
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: Remove "please"

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 2]
```diff
@@ -69,21 +70,40 @@ the following changes are required:
...
+   :file:`home/vagrant/.ssh/` and :file:`home/vagrant/.ssh/authorized_keys`
+   correctly, otherwise vagrant will not be able to connect to your
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: vagrant -> Vagrant

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 3]
```diff
@@ -69,21 +70,40 @@ the following changes are required:
...
+5. Create the default shared folder
+
+   Vagrant boxes usually provide a default shared folder under
+   :file:`/vagrant`. Consider adding this empty folder to your overlay
+   files and ensure that the user ``vagrant`` has write permissions for
+   it.
+
+6. Setup and start SSH daemon
+
+   In :file:`config.sh` add the start of sshd and the initial creation of
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: Add comma after config.sh.

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 4]
```diff
@@ -97,15 +117,20 @@ the following changes are required:
       #--------------------------------------
       suseInsertService sshd
 
-   Also make sure to setup **UseDNS=no** in :file:`etc/ssh/sshd_config`
-   This can be done by an overlay file or clever patching of
-   the file in the above mentioned :file:`config.sh` file.
+   Also make sure to setup **UseDNS=no** in :file:`/etc/ssh/sshd_config`
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: "to setup ..." as a phrase does not make sense to me. Either you **set** a value to a key, or you add a line. Here are the alternatives:

* Also make sure to set ...
* Also make sure to add the line ...

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 5]
```diff
@@ -97,15 +117,20 @@ the following changes are required:
...
+   This can be done by an overlay file or by patching the file in the above
+   mentioned :file:`config.sh` file.
 
-6. Configure sudo for Vagrant user
+7. Configure sudo for the vagrant user
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: vagrant -> Vagrant

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 6]
```diff
@@ -97,15 +117,20 @@ the following changes are required:
...
+
+   Vagrant expects to have passwordless root permissions via ``sudo`` to be
+   able to setup your box. Please add the following line to
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: Remove "please"

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 7]
```diff
@@ -97,15 +117,20 @@ the following changes are required:
...
+   :file:`/etc/sudoers` to ensure this:
+
+   .. code::
+
+      vagrant ALL=(ALL) NOPASSWD: ALL
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: Hmn, shouldn't the line be added with the command `visudo` instead of adding it with an ordinary editor?

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 8]
```diff
@@ -134,3 +160,7 @@ with the following sequence of :command:`vagrant` commands:
    Currently, KIWI only supports the libvirt provider. There are
    other providers like virtualbox and also vmware available which
    requires a different box layout currently not supported by KIWI.
+
+
+.. [#f1] The insecure key is removed from the box when the it is first
+         booted via vagrant.
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: Consistency: vagrant -> Vagrant

File: doc/source/index.rst [hunk 9]
```diff
@@ -29,14 +29,45 @@ Appliance ?
...
 provided as an image file and needs to be deployed to, or activated in
 the target system or service.
 
-In KIWI the appliance is specified by a collection of human readable files
-in a directory, also called as the `image description`. At least one XML
-file :file:`config.xml` or :file:`.kiwi` is required. In addition
-there may be as well other files like scripts or configuration data.
+KIWI can create appliances in various forms: beside classical installation
+ISOs and images for virtual machines it can also build images that boot via
+PXE or vagrant boxes.
```
Comment (tomschr) [inline doc/source/index.rst]: Consistency: vagrant -> Vagrant

File: doc/source/index.rst [hunk 10]
```diff
@@ -29,14 +29,45 @@ Appliance ?
...
+
+In KIWI the appliance is specified via a collection of human readable files
+in a directory, also called the `image description`. At least one XML file
+:file:`config.xml` or :file:`.kiwi` is required. In addition there may as
+well be other files like scripts or configuration data.
+
+
+Use cases
```
Comment (tomschr) [inline doc/source/index.rst]: Use title style: Use Cases

File: doc/source/index.rst [hunk 11]
```diff
@@ -29,14 +29,45 @@ Appliance ?
...
+---------
+
+Not convinced yet? You can find a selection of the possible uses cases
+below:
+
+* You are a system administrator and wish to create a customized installer
+  for your network that includes additional software and your organizations
+  certificates? KIWI allows you to select which packages will be included
+  in the final image. On top of that you can add files to arbitrary
+  locations in the filesystem, e.g. to include ssl or ssh keys. You can
```
Comment (tomschr) [inline doc/source/index.rst]: * e. g. -> for example, 
* ssl -> SSL
* ssh -> SSH

File: doc/source/index.rst [hunk 12]
```diff
@@ -29,14 +29,45 @@ Appliance ?
...
+  also tell KIWI to create an image that can be booted via PXE, so that you
+  don't even have to leave your desk to reinstall a system.
+
+* You want to create a custom spin of your favorite Linux distribution with
+  additional repositories and packages that are not present by default?
+  With KIWI you can configure the repositories of your final image via the
+  image description and tweak the list of packages that are going to be
+  installed to match your target audience.
+
+* The Raspberry Pi that is coordinating your home's IoT devices got very
```
Comment (tomschr) [inline doc/source/index.rst]: An unknown acronym should be explained at least once: ... your home's Internet of Thing (IoT) devices...

File: doc/source/quickstart.rst [hunk 13]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
+   This document describes how to start working with KIWI, an OS appliance
+   builder.
    This description applies for version |version|.
 
 Before you start
 ----------------
 
-Make sure you have installed KIWI and the example collection as
-explained in :ref:`kiwi-installation`
+1. Please install KIWI first, either via your distributions' package
```
Comment (tomschr) [inline doc/source/quickstart.rst]: Remove "please"

File: doc/source/quickstart.rst [hunk 14]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
+
+2. Clone the repository containing example appliances (see
+   :ref:`example-descriptions`):
+
+   .. code:: bash
+
+      $ git clone https://github.com/SUSE/kiwi-descriptions
+
+
+Choose a first image
```
Comment (tomschr) [inline doc/source/quickstart.rst]: Title style: Choose a First Image

File: doc/source/quickstart.rst [hunk 15]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
 
     $ sudo kiwi-ng --type vmx system build \
         --description kiwi-descriptions/suse/x86_64/suse-leap-42.3-JeOS \
         --target-dir /tmp/myimage
 
-Find the image with the suffix :file:`.raw` below :file:`/tmp/myimage`.
+The resulting image will be in placed into the folder :file:`/tmp/myimage`
+with the suffix :file:`.raw`.
+
+If you don't wish to create a Leap 42.3 image, simply substitute the folder
```
Comment (tomschr) [inline doc/source/quickstart.rst]: * Leap -> openSUSE Leap
* Remove "simply"

File: doc/source/quickstart.rst [hunk 16]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
+following the ``--description`` option with another folder that contains
+the image description which you selected.
 
 Run your image
```
Comment (tomschr) [inline doc/source/quickstart.rst]: Title style: Run Your Image

File: doc/source/quickstart.rst [hunk 17]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
+will be using QEMU and boot it as follows:
 
 .. code:: bash
 
     $ qemu \
         -boot c
         -drive file=LimeJeOS-Leap-42.3.x86_64-1.42.3.raw,format=raw,if=virtio \
         -m 4096
+
+Tweak and customize your image
```
Comment (tomschr) [inline doc/source/quickstart.rst]: Title style: Tweak and Customize your Image

File: doc/source/quickstart.rst [hunk 18]
```diff
@@ -5,40 +5,75 @@ Quick Start
...
+------------------------------
+
+Now that you have successfully built and started your first image, you can
+start tweaking it to match your needs.
+
+Please find the documentation of the appliance description files in the
```
Comment (tomschr) [inline doc/source/quickstart.rst]: Remove "please"

**Comment by tomschr** [comment]
@dcermak 
> I'll proofread the whole thing once more today or on Monday, so don't bother checking this before I mark it as ready for review.

Uups, sorry. I saw your comment when I was already finished.

File: doc/source/building/build_in_buildservice.rst [hunk 19]
```diff
@@ -9,19 +9,157 @@ Building in the Open Buildservice
...
+  </repository>
+
+
+Due to the nature of OBS' dependency resolution mechanism a lot of
+``Prefer:``, ``Substitute:``, ``Preinstall:`` and ``Support:`` directives
+are required for an image to build successfully without having choices or
+conflicts for packages. We therefore recommend to initially copy the
+current project configuration of the testing project
+`Virtualization:Appliances:Images:Testing_$ARCH` into your own project as a
+start and tweak it from there instead of starting from a blank slate.
```
[the remark this answers is not in the document]
Comment (dcermak) [reply doc/source/building/build_in_buildservice.rst]: Hm "blank state" sounds odd to be honest. This sentence should just say that this configuration will provide the user with a better initial version than starting at zero. That's why I wrote blank slate, but I'm not a native speaker, so if there's a better phrase: I'm open for suggestions.

File: doc/source/building/working_with_images/vagrant_setup.rst (hunk 7 above)
Comment (dcermak) [reply doc/source/building/working_with_images/vagrant_setup.rst]: Can you do that non-interactively? This whole enumeration describes steps that one has to add to `config.sh` so that Vagrant will be able to boot the box. (If that is not clear from the text, please let me know!)

**Comment by dcermak** [comment]
@tomschr No problem, I fixed most of your suggestions and corrected some other parts too.

File: doc/source/building/build_in_buildservice.rst [hunk 20]
```diff
@@ -1,27 +1,166 @@
...
+build without modifications.
+
+The notable differences to running KIWI locally include:
+
+* OBS will pick the KIWI package from the available repositories that you
+  specified in :file:`config.xml`/:file:`config.kiwi`. This means, that you
+  will be very likely building with a different KIWI version in the Open
+  Build Service.
+  This is especially relevant when building images for older versions
+  like SLE. Therefore, include the custom appliances repository as
```
Comment (tomschr) [inline doc/source/building/build_in_buildservice.rst]: Does everybody knows this SLE acronym? Maybe you should better write the complete name (like SUSE Linux Enterprise).

File: doc/source/building/build_in_buildservice.rst [hunk 21]
```diff
@@ -1,27 +1,166 @@
...
+
+  Place the above line into the project configuration, which can be
+  accessed either via the web interface (click on the tab ``Project
+  Config`` on your project's main page) or via ``osc meta -e prjconf``.
+
+* OBS can only build a single image type, for example it cannot build a VMX
+  disk and a live ISO from one :file:`config.xml`. A configuration file
+  that contains multiple image types will therefore not work in the Open
+  Build Service.
+  Split the configuration file into two (or more) and provide them as
```
Comment (tomschr) [inline doc/source/building/build_in_buildservice.rst]: Split into two (or more)... files?

File: doc/source/building/build_in_buildservice.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (tomschr) [reply doc/source/building/build_in_buildservice.rst]: Well, I'm not a native speaker as well, but I considered "slate" more of a typo than an appropriate word. :wink: However, I may be wrong.

Maybe we should rephrase it to avoid this word? What about the following sentences:

```
... into your own project as a start and tweak it from there instead of starting from scratch.
... into your own project as a start and tweak it from there instead of starting anew.
```

File: doc/source/building/working_with_images/vagrant_setup.rst [hunk 22]
```diff
@@ -69,21 +70,40 @@ the following changes are required:
...
+   name of the user as the password for login.
 
 4. Integrate public SSH key
 
-   Reach out to
-   `Insecure Keys <https://github.com/hashicorp/vagrant/tree/master/keys>`_
-   for details on this keys. Add the public key to the
-   box as overlay file in your image description at
-   :file:`home/vagrant/.ssh/authorized_keys`
+   Vagrant requires an insecure public key pair [#f1]_ to be added to the
```
Comment (tomschr) [inline doc/source/building/working_with_images/vagrant_setup.rst]: No space between the word and the footnote. So it should be "key pair¹" instead of "key pair ¹".

File: doc/source/building/working_with_images/vagrant_setup.rst (hunk 7 above)
Comment (tomschr) [reply doc/source/building/working_with_images/vagrant_setup.rst]: > Can you do that non-interactively?

I'm not sure if that is possible.  Maybe `visudo` is not really needed, but `visudo` advertise itself as "visudo - safely edit the sudoers file". So I thought, it *might* be something useful.

> (If that is not clear from the text, please let me know!)

It was clear to me, so no worries. :wink: 

----

By the way, just as a sidenote: would it make sense to create a file under `/etc/suders.d/` instead of manipulating the file `/etc/sudoers` directly?
One benefit could be, that all the configuration is in *one* file. If you don't need it anymore, you can remove it quite easily.

File: doc/source/index.rst [hunk 23]
```diff
@@ -29,14 +29,45 @@ Appliance ?
...
 
-In KIWI the appliance is specified by a collection of human readable files
-in a directory, also called as the `image description`. At least one XML
-file :file:`config.xml` or :file:`.kiwi` is required. In addition
-there may be as well other files like scripts or configuration data.
+KIWI can create appliances in various forms: beside classical installation
+ISOs and images for virtual machines it can also build images that boot via
+PXE or Vagrant boxes.
+
+In KIWI the appliance is specified via a collection of human readable files
```
Comment (tomschr) [inline doc/source/index.rst]: Comma after "In KIWI".

File: doc/source/quickstart.rst [hunk 24]
```diff
@@ -5,40 +5,76 @@ Quick Start
...
 
-Run your image
+Run your Image
 --------------
 
 Running an image actually means booting the operating system. In order
-to do that attach the disk image to a virtual system, in our case QEMU
-is used, and boot it as follows:
+to do that attach the disk image to a virtual system. In this example we
+will be using QEMU and boot it as follows:
```
Comment (tomschr) [inline doc/source/quickstart.rst]: will be using -> use
(most documentation uses present tense)

File: doc/source/building/working_with_images/vagrant_setup.rst (hunk 22 above)
Comment (dcermak) [reply doc/source/building/working_with_images/vagrant_setup.rst]: The space is mandatory in sphinx: the footnote will not be resolved when the space is missing.

Comment (tomschr) [reply doc/source/building/working_with_images/vagrant_setup.rst]: You're absolutely right. Sorry, ignore my idea.

**Comment by dcermak** [comment]
@tomschr Can't reply to your comment about `/etc/sudoers.d`: I have included that in the text and added a check using `visudo`.

File: doc/source/quickstart.rst [hunk 25]
```diff
@@ -5,40 +5,76 @@ Quick Start
...
+
+Choose a First Image
+--------------------
+
+Take a look which images are available in the example appliances repository
+and select one that matches your desired image as close as possible. Or
+just use the one given in the examples below.
+
 
 Build your first image
```
Comment (tomschr) [inline doc/source/quickstart.rst]: For consistency reasons, use title style too: Build your First Image

File: doc/source/building/build_in_buildservice.rst [hunk 26]
```diff
@@ -1,27 +1,167 @@
...
+
+* OBS can only build a single image type, for example it cannot build a VMX
+  disk and a live ISO from one :file:`config.xml`. A configuration file
+  that contains multiple image types will therefore not work in the Open
+  Build Service.
+  Split the configuration file into two (or more) :file:`.kiwi` files and
+  provide them as separate packages inside your project.
+
+* OBS projects do not support subfolders, thus you must provide the overlay
+  files packed as an archive called :file:`root.tar.gz`.
```
Comment (Conan-Kudo) [inline doc/source/building/build_in_buildservice.rst]: There's actually one more very important restriction:

* KIWI image builds in OBS cannot use `namedCollection` items, as the metadata backing those are not available. This is quite obvious when you try to use it to install groups for RH/Fedora image builds.

File: doc/source/installation.rst [hunk 27]
```diff
@@ -6,18 +6,18 @@ Installation
...
 installation as a packages through a package manager is described.
 
-Packages for the new KIWI version are provided at the `openSUSE
+Packages for the new KIWI version are provided by the `openSUSE
 buildservice <http://download.opensuse.org/repositories/Virtualization:/Appliances:/Builder>`__.
 
 To install KIWI, do:
 
-1. Open the URL http://download.opensuse.org/repositories/Virtualization:/Appliances:/Builder
+1. Open the URL https://download.opensuse.org/repositories/Virtualization:/Appliances:/Builder
```
Comment (Conan-Kudo) [inline doc/source/installation.rst]: It is also packaged in Fedora and installable by just running `dnf install kiwi-cli` (technically `dnf install kiwi` works too, since the `kiwi` name is provided by `kiwi-cli`).

File: doc/source/building/build_in_buildservice.rst (hunk 26 above)
Comment (schaefi) [reply doc/source/building/build_in_buildservice.rst]: I would add something like

```
* OBS projects do not support the use of the namedCollection section.
  A specification of ``` <namedCollection name="collection_name"/>``` is used to pass the
  information to install a collection of packages to the used package manager. The package
  manager can resolve that information only if the repository metadata contains the information
  about that collection and its packages. If the buildservice builds the image it resolves the given
  package list using its own sat solver based solver. From the result the buildservice creates
  temporary repositories of the same names as they got configured in the kiwi XML description.
  Those repositories doesn't contain the metadata to resolve collections.

  Even though kiwi uses the package manager to register the repositories, one should understand
  that the repositories the buildservice creates are just a subset of the data the real
  repositories provides.

  The dependency resolver mechanism the buildservice applies to create that repositories before
  kiwi is called is also different compared to the dependency resolution the selected package 
  manager performs later when kiwi calls it:

  * In the buildservice the order of repositories in the XML description matters
  * In the buildservice package dependencies from file provides are not resolved

  Because of that reason an image build that resolves well outside of the buildservice might
  not resolve when moving the image build into the buildservice
```

Comment (schaefi) [reply doc/source/building/build_in_buildservice.rst]: Maybe this can be written nicer, or in sub paragraphs but I think that information is good to have here

Comment (dcermak) [reply doc/source/building/build_in_buildservice.rst]: I have added the text in a slightly modified form: I have split up some of the sentences and formulated some of them differently.

Could you double check that it's still correct? I don't really understand most of that (but the behavior sound pretty, let's call it, odd tbh...).

File: doc/source/installation.rst (hunk 27 above)
Comment (dcermak) [reply doc/source/installation.rst]: I haven't found a good place where to fit that in, so I extended the installation section and added two additional sections: one for installing from pip and one to install from your distro's package manager.

File: doc/source/building.rst [hunk 28]
```diff
@@ -68,8 +68,8 @@ which are **equal** or **newer** compared to the following list:
 *  openSUSE Leap 42
 *  Red Hat Enterprise 7
 *  SUSE Linux Enterprise 12
-*  Tumbleweed
-*  Ubuntu Xenial
+*  openSUSE Tumbleweed
+*  Ubuntu Xenial Xerus
```
Comment (Conan-Kudo) [inline doc/source/building.rst]: Change this to `16.04 (Xenial Xerus)`.

File: doc/source/installation.rst [hunk 29]
```diff
@@ -6,51 +6,118 @@ Installation
...
+repositories. These include openSUSE Tumbleweed, openSUSE Leap and Fedora
+since version 28.
+
+To install KIWI on openSUSE, run the following command:
+
+.. code-block:: shell-session
+
+   $ sudo zypper install python3-kiwi
+
+On Fedora use the following command instead:
```
Comment (Conan-Kudo) [inline doc/source/installation.rst]: `On Fedora use` -> `On Fedora, use`

**Comment by Conan-Kudo** [comment]
@schaefi I think @dcermak already did. The commits are logically separated.

**Comment by schaefi** [comment]
yes sorry, my fault. Looks good. Thanks
